### PR TITLE
1370 making comments live

### PR DESCRIPTION
**Pull request by zimboflyman**
### What is the context of this PR?
Added a subsrciption to the comments on question and Calculated summary pages.
This will automatically re-render for other logged in users when my user makes any change to comments.

### How to review 
On a Question page in Preview Tab:
1/ Log in as any user (A)
2/ open an incognito chrome window and login as a different user (B)
3/ make the following changes to comments: 
New comment, 
Edit comment, 
Delete comment
4/ you should see user B automatically updating the same page to show your comments.

Repeat for Calculated Summary Page

### CODE REVIEW

**Comment by zimboflyman** [comment]
[Making comments live](https://trello.com/c/yhRpfs8D/1370-making-comments-live)

File: eq-author/src/components/CommentsPanel/index.js [hunk 1]
```diff
@@ -12,6 +14,7 @@ import COMMENT_QUERY from "./commentsQuery.graphql";
 import COMMENT_ADD from "./createNewComment.graphql";
 import COMMENT_DELETE from "./deleteComment.graphql";
 import COMMENT_UPDATE from "./updateComment.graphql";
+// import COMMENT_SUBSCRIPTION from "./commentsSubscription.graphql";
```
Comment (jasehumphr) [inline eq-author/src/components/CommentsPanel/index.js]: // should remove this

File: eq-author/src/components/CommentsPanel/index.test.js [hunk 2]
```diff
@@ -335,6 +386,8 @@ describe("Comments Pane", () => {
     await flushPromises();
 
     expect(deleteWasCalled).toBeTruthy();
+    // expect(newCommentSubscriptionWasCalled).toBeTruthy();
```
Comment (jasehumphr) [inline eq-author/src/components/CommentsPanel/index.test.js]: // should remove this comment

File: eq-author-api/schema/tests/subscription.test.js [hunk 3]
```diff
@@ -181,6 +187,181 @@ describe("subscriptions", () => {
...
+        commentId: commentId,
+      });
+      const result = await iterator.next();
+      expect(result.value.data.commentsUpdated).toMatchObject({
+        id: questionPageId,
+        comments: [],
+      });
+    });
+
+    //////////
```
Comment (heofs) [inline eq-author-api/schema/tests/subscription.test.js]: I think this should be removed

File: eq-author-api/schema/typeDefs.js [hunk 4]
```diff
@@ -1098,9 +1098,11 @@ input DeleteCollapsibleInput {
   id: ID!
 }
 
+
```
Comment (heofs) [inline eq-author-api/schema/typeDefs.js]: too much space :)

File: eq-author-api/yarn.lock [hunk 5]
```diff
@@ -991,10 +991,11 @@ apollo-link@^1.2.3:
     zen-observable-ts "^0.8.19"
 
 apollo-opentracing@latest:
-  version "1.4.1"
-  resolved "https://registry.yarnpkg.com/apollo-opentracing/-/apollo-opentracing-1.4.1.tgz#878781ed8e8abf3b257122991fd55b5f1e219ea9"
-  integrity sha512-UrRvEnXQMoV8StdRyYiVEkJoOCviO6O9xCZGwPVRxLsj3VlwmFr3yeaf2M/PuJgwoEXq7h2SX0JmMLrm/EUdfA==
+  version "1.4.2"
+  resolved "https://registry.yarnpkg.com/apollo-opentracing/-/apollo-opentracing-1.4.2.tgz#2c96af989685a4a72a173e683b8dd11fe7c79b31"
+  integrity sha512-5IgBIixqqmWWOo+TAsPYVRq9JbUgoPnRLckutgIgFpoSVxhM78nkAmwuXBF8lRveXWy113GnDEihAJ2nuOmF2Q==
```
Comment (heofs) [inline eq-author-api/yarn.lock]: is package update necessary here? I suspect this has already been updated too.

File: eq-author/src/components/CommentsPanel/index.js [hunk 6]
```diff
@@ -229,6 +231,33 @@ const CommentsPanel = ({
...
+            picture
+            email
+            displayName
+          }
+          createdTime
+          editedTime
+        }
+      }
+    }
+  `;
```
Comment (heofs) [inline eq-author/src/components/CommentsPanel/index.js]: This should be outside the component. It should probably be moved to its own file too.

This part can probably be moved into a fragment if it is used in multiple places.
```graphql
        id
        comments {
          id
          commentText
          user {
            id
            name
            picture
            email
            displayName
          }
          createdTime
          editedTime
        }
```

File: eq-publisher/yarn.lock [hunk 7]
```diff
@@ -1589,9 +1589,9 @@ eslint-visitor-keys@^1.1.0:
   integrity sha512-8y9YjtM1JBJU/A9Kc+SbaOV4y29sSWckBwMHa+FGtVj5gN/sbnKDf6xJUl+8g7FAij9LVaP8C24DUiH/f/2Z9A==
 
 eslint@latest:
-  version "6.7.1"
-  resolved "https://registry.yarnpkg.com/eslint/-/eslint-6.7.1.tgz#269ccccec3ef60ab32358a44d147ac209154b919"
-  integrity sha512-UWzBS79pNcsDSxgxbdjkmzn/B6BhsXMfUaOHnNwyE8nD+Q6pyT96ow2MccVayUTV4yMid4qLhMiQaywctRkBLA==
+  version "6.7.2"
+  resolved "https://registry.yarnpkg.com/eslint/-/eslint-6.7.2.tgz#c17707ca4ad7b2d8af986a33feba71e18a9fecd1"
+  integrity sha512-qMlSWJaCSxDFr8fBPvJM9kJwbazrhNcBU3+DszDW1OlEwKBBRWsJc7NJFelvwQpanHCR14cOLD41x8Eqvo3Nng==
```
Comment (heofs) [inline eq-publisher/yarn.lock]: I think some changes from another PR has been pulled in here. I dont think any changes to dependencies are needed for this PR

Comment (zimboflyman) [reply eq-publisher/yarn.lock]: I did do a rebase today - so that will explain it...

File: eq-author-api/yarn.lock (hunk 5 above)
Comment (zimboflyman) [reply eq-author-api/yarn.lock]: caused by rebase

File: eq-author-api/schema/tests/comments.test.js [hunk 8]
```diff
@@ -73,6 +73,34 @@ describe("comments", () => {
...
+      pageId: createdQuestionPage.id,
+      commentText: "a new comment is created",
+    });
+
+    await createComment(ctx, {
+      pageId: createdQuestionPage.id,
+      commentText: "a 2nd comment is created",
+    });
+
+    const queryNewComment = await queryComments(ctx, {
```
Comment (heofs) [inline eq-author-api/schema/tests/comments.test.js]: change this const to `queryNewComments` as it is is multiple of them.

File: eq-author/src/App/page/Preview/QuestionPagePreview.test.js [hunk 9]
```diff
@@ -20,6 +21,27 @@ describe("QuestionPagePreview", () => {
 
   actSilenceWarning();
 
+  const commentsSubscription = gql`
```
Comment (heofs) [inline eq-author/src/App/page/Preview/QuestionPagePreview.test.js]: You could import the query from this file `eq-author/src/components/CommentsPanel/commentSubscription.graphql`

File: eq-author/src/App/page/Preview/CalculatedSummaryPreview.test.js [hunk 10]
```diff
@@ -14,6 +15,27 @@ describe("CalculatedSummaryPreview", () => {
 
   actSilenceWarning();
 
+  const commentsSubscription = gql`
+    subscription CommentsUpdated($pageId: ID!) {
```
Comment (heofs) [inline eq-author/src/App/page/Preview/CalculatedSummaryPreview.test.js]: You could import the query from this file `eq-author/src/components/CommentsPanel/commentSubscription.graphql`

File: eq-author-api/schema/tests/subscription.test.js [hunk 11]
```diff
@@ -17,6 +17,12 @@ const {
 const {
   updateQuestionPage,
 } = require("../../tests/utils/contextBuilder/page/questionPage");
+const {
+  // queryComments,
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/schema/tests/subscription.test.js]: woops - done!

File: eq-author-api/schema/tests/comments.test.js (hunk 8 above)
Comment (zimboflyman) [reply eq-author-api/schema/tests/comments.test.js]: makes sense - ta

File: eq-author/src/App/page/Preview/CalculatedSummaryPreview.test.js [hunk 12]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import { shallow } from "enzyme";
 import { render, flushPromises } from "tests/utils/rtl";
 import actSilenceWarning from "tests/utils/actSilenceWarning";
+import commentsSubscription from "../../../components/CommentsPanel/commentSubscription.graphql";
```
Comment (heofs) [inline eq-author/src/App/page/Preview/CalculatedSummaryPreview.test.js]:
```suggestion
import commentsSubscription from "components/CommentsPanel/commentSubscription.graphql";
```

All folders in `eq-author/src/` can be imported like that. Not the same for `eq-author-api` and `eq-publisher` though.

File: eq-author/src/App/page/Preview/QuestionPagePreview.test.js [hunk 13]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import { shallow } from "enzyme";
 import { render, flushPromises } from "tests/utils/rtl";
 import actSilenceWarning from "tests/utils/actSilenceWarning";
+import commentsSubscription from "../../../components/CommentsPanel/commentSubscription.graphql";
```
Comment (heofs) [inline eq-author/src/App/page/Preview/QuestionPagePreview.test.js]:
```suggestion
import commentsSubscription from "/components/CommentsPanel/commentSubscription.graphql";
```

File: eq-author/src/components/CommentsPanel/index.js [hunk 14]
```diff
@@ -1,5 +1,6 @@
 import React, { useState, useRef, useEffect } from "react";
-import { withRouter } from "react-router-dom";
+import { withRouter } from "react-router";
```
Comment (heofs) [inline eq-author/src/components/CommentsPanel/index.js]: Should be `react-router-dom`

**Pull request merged**